server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -172,17 +173,14 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
         const req = async () => {
             const result = await this.collection.findOneAndUpdate(
                 query,
-                {
-                    $setOnInsert: value,
-                },
-                {
-                    returnOriginal: true,
-                    upsert: true,
-                });
+                {setOnInsert: value},
+                {upsert: true,
+                returnDocument: "before"}
+            );
 
             return result.value
-                ? { value: result.value, existing: true }
-                : { value, existing: false };
+                ? { value: result.value, existing: true } as { value: T; existing: boolean; }
```

Comment:
@znewton Need some suggestions to handle this sentence. Thanks. :)

Otherwise, it would give us this error:
<img width="1109" alt="Screenshot 2023-03-01 at 3 46 12 PM" src="https://user-images.githubusercontent.com/66701969/222293175-0394d93c-b950-40b6-9f95-f204cb1a8a85.png">

Reply:
I have this error when I would like to adapt the above code. I think the type in new MongoDB drive changes to withId<T>
<img width="980" alt="Screenshot 2023-03-01 at 4 05 52 PM" src="https://user-images.githubusercontent.com/66701969/222295658-bccb7373-fc1d-462c-bc79-a74b0fede640.png">